### PR TITLE
Bbox documentation

### PR DESCRIPTION
**Pull request by DenisCarriere**
First stab at handling GeoJSON BBox & Google's Bounds input.

@ebreton Might be worth having a helper method to handle BBox across all providers that support BBox.

### CODE REVIEW

**Comment by ebreton** [comment]
Hi @DenisCarriere , 

I followed on your first stab 👍 and your suggestion to add a helper for bbox: it's a new class in location.py with a factory (I have added the appropriate tests also, which give some examples too)

-> I have adapted consequently the three providers that offer a proximity query: 
- geonames
- google place
- mapbox 

The new BBox class could surely be used in the other cases where a bbox is implied, but I prefer that you have a look at it beforehands, and see for yourself if that worth adapting further.

The doc is also reviewed with all this (and also the api.py docstrings). As usual: feel free to enrich / correct.

File: docs/results.rst [hunk 1]
```diff
@@ -42,7 +42,7 @@ Extending without breaking
 The objective is to allow access to multiple results, without breaking the lib, neither adding to much complexity.
 
 - all existing API calls shoul continue to Work
-- access to all restults should be easy
```
Comment (DenisCarriere) [inline docs/results.rst]: 🤦‍♂️

File: docs/results.rst [hunk 2]
```diff
@@ -147,3 +146,87 @@ The returned object *g* is a `MutableSequence (python >3.3) <https://docs.python
...
+    >>> bbox = {
+            'southwest': [43.0, -80.5],
+            'northeast': [43.5, -80.0]
+        }
+    >>> g = geocoder.geonames('Paris', proximity=bbox, key='<USERNAME>')
+    >>> print([g.address, g.country, g.latlng])
+    ['Paris', 'Canada', ['43.2001', '-80.38297']]
+
+    # let's do the same with bounds
+    >>> bounds = [-80.5, 43.0, -80.0, 43.5]
```
Comment (DenisCarriere) [inline docs/results.rst]: Bounds & BBox would be reversed, `bbox` is an Array of 4 numbers.

File: geocoder/geonames.py [hunk 3]
```diff
@@ -94,11 +95,12 @@ def _build_params(self, location, provider_key, **kwargs):
             'maxRows': kwargs.get('maxRows', 1),
         }
         # check out for bbox in kwargs
-        bbox = kwargs.pop('bbox', None)
+        bbox = kwargs.pop('proximity', None)
```
Comment (DenisCarriere) [inline geocoder/geonames.py]: 👍 That's way better using `proximity` instead of a specific `bbox` or `bounds` term.

File: geocoder/location.py [hunk 4]
```diff
@@ -121,6 +127,90 @@ def __str__(self):
...
+            return cls(bounds=arg)
+        else:
+            raise ValueError(
+                "Could not found valid values in list to create a bbox")
+
+    def __init__(self, bbox=None, bounds=None,
+                 lat=None, lng=None,
+                 west=None, south=None, east=None, north=None):
+        if bbox is not None:
+            self.south, self.west = map(float, bbox['southwest'])
```
Comment (DenisCarriere) [inline geocoder/location.py]: Just to keep things consistent, we should reverse `bbox` & `bounds` (bbox = 4 number Array).

@ebreton I'll push a commit, but you might need to look over it again to make sure it's good.

**Comment by DenisCarriere** [comment]
> What about adding some columns in the the README.

👍 That sounds like a good plan

**Comment by ebreton** [comment]
actually... done ;)

**Comment by DenisCarriere** [comment]
Oh perfect! 👍

**Pull request merged**